File: 백준/Bronze/1259. 팰린드롬수/팰린드롬수.java
import java.util.*;
public class Main {
    public static void main(String[] args) {
        // 여기에 코드를 작성해주세요.
        Scanner sc = new Scanner(System.in);

        int n;
       
        
        for(int k = 0; k < 3000; k++){
            n = sc.nextInt();
            boolean pcheck = true;
            if(n==0)
            break;
            String num = Integer.toString(n);
            int len = num.length();
            char[] arr = new char[len];

            for(int i = 0; i < len; i++){
                arr[i] = num.charAt(i);
            }
        
            for(int i = 0; i<len/2; i++){
                if(arr[i]!=arr[len-1-i])
                pcheck = false;
            }

            if(pcheck == true)
            System.out.println("yes");
            else
            System.out.println("no");
        }
    }
}

File: 백준/Bronze/1259. 팰린드롬수/팰린드롬수.py
arr = []

while True:
    n = input()
    if n == '0':
        break
    else:
        arr.append(n)

for n in arr:
    p = True
    for i in range(len(n)//2):
        if n[i] != n[-1-i]:
            p = False
            break
    print("yes" if p else "no")